yarn_api_client/base.py

```diff
@@ -23,20 +41,9 @@ def __init__(self, service_endpoint=None, timeout=None, auth=None, verify=True):
         self.timeout = timeout
 
         if service_endpoint:
-            # Snippet to decouple parts of Uri for further building of URLs
-            self.service_uri = urlparse(service_endpoint)
-            self.service_uri_scheme = self.service_uri.scheme or 'http'
-            self.service_uri_hostname = self.service_uri.hostname or self.service_uri.path
-
-            if self.service_uri.port:
-                self.service_uri_postfix = ':' + str(self.service_uri.port)
-            else:
-                self.service_uri_postfix = ''
+            self.service_uri = Uri(service_endpoint)
         else:
             self.service_uri = None
```

Comment:
Mainly due to existing test with indirect variable assignment in here:
https://github.com/toidi/hadoop-yarn-api-python-client/pull/34/files#diff-d06796e77090d59038f12fe9b1260baeL62
However, I do completely agree that it makes sense to validate right in constructor and rewrite test to use proper constructor case (since its meant to be used like this afterall!). Let me know what you think.

Reply:
Thanks for the link.  So I suppose the original intent was that clients could update the connection information after creating the instance.  That seems highly problematic to me - especially from a supportability standpoint.  However, because this is python, they'd still be able to do that, so I see the need to retain `_validate_configuration()` and, because this methodology may be in use (i.e., using the instance more like a singleton to hit different YARN clusters), I'm thinking it might be best to leave things as they are.